src/components/file-upload/index.md.njk

```diff
@@ -23,6 +23,54 @@ There are 2 ways to use the file upload component. You can use HTML or, if you
 
 {{ example({group: "components", item: "file-upload", example: "default", html: true, nunjucks: true, open: false, id: "default-2"}) }}
 
+### Error messages
+
+Check that a valid date is entered by the user. Invalid dates should be reported to the user like this:
+
+{{ example({group: "components", item: "file-upload", example: "error", html: true, nunjucks: true, open: false, size: "m"}) }}
+
+Make sure errors follow the guidance in [error message](/components/error-message/) and have specific error messages for specific error states.
+
+#### If nothing has been uploaded
+Say ‘Select a [whatever they need to upload]’.<br>
+For example, ‘Select a report’.
+
+#### If the file is the wrong file type
+Say‘The [whatever] must be a [list of file types]’.<br>
+For example, ‘The report must be a CSV or ODS’ or ‘The upload must be a JPG, BMP, PNG, TIF or PDF’.
+
+#### If the file is too big
+Say ‘The [whatever] must be smaller than [largest file size]’.<br>
+For example, ‘The CSV must be smaller than 2MB’.
+
+#### If the file is empty
+Say ‘The [whatever] is empty – select a [whatever] with data in it’.<br>
+For example, ‘The CSV is empty – select a CSV with data in it’.
+
+#### If the file contains a virus
+Say ‘The [whatever] contains a virus’.<br>
+For example, ‘The CSV contains a virus’.
+
+#### If the file is password protected
+Say ‘The [whatever] is password protected’.<br>
+For example, ‘The report is password protected’.
+
+#### If there was a problem and the file was not uploaded
+Say ‘The [whatever] could not be uploaded – try again’.<br>
+For example, ‘The CSV could not be uploaded – try again’.
+
+#### If the user can upload more than one file but there is a limit on how many they can upload
+Say ‘You can only upload up to [highest number] [whatevers] at the same time’.<br>
+For example, ‘You can only upload up to 10 reports at the same time’.
+
+#### If the upload must be in a template that has been changed and you do know what has changed
```

Comment:
I find this heading (and the next one) really hard to parse. I think it's mainly because of the use of 'upload' as a noun, coupled with the phrase 'in a template' which I don't think really makes sense?

Reply:
I agree. I have tried to make them better.

Reply:
I'm still finding these quite clunky and difficult to read. I'm also just wondering if this level of specificity is necessary? How common a use case actually is this? If we were to hold this up to the Design System criteria – is there evidence that multiple services will be needing (and able) to scan uploaded files to check for changes? I'd honestly be tempted to leave these out – what do you think @stevenaproctor?

Reply:
I know of at least two services in HMRC that has templates that people must use for uploads. But the templates are very different so these kinds of errors are had to create templates for.

I will remove these two and leave the general template one at the bottom as a starter.